text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,852 @@
+
+See [#3966](https://github.com/open-telemetry/opentelemetry-specification/issues/3966) for context on this issue.
+
+While details of ENV variables will be subject to change, it would look something like the following:
+
+```bash
+set OTEL_DETECTED_ENTITIES=k8s.deployment[k8s.deployment.name=my-program],k8s.pod[k8s.pod.name=my-program-2314,k8s.namespace=default]
+<run my program>
+```
+
+The minimum requirements of this entity detector are:
+
+- ENV variable(s) can specify multiple entities (resource attribute bundles)
+- ENV variable(s) can be easily appended or leverages by multiple participating systems, if needed.
+- Entities discovered via ENV variable(s) can participate in Resource Provider generically, i.e. resolving conflicting definitions.
+- ENV variable(s) have a priority that can be influenced by platform entity providers (e.g. prepending vs. appending)
+
+The actual design for this ENV variable interaction would follow the approval of this OTEP.
+
+### Interactions with OpenTelemetry Collector
+
+The OpenTelemetry collector can be updated to optionally  interact with Entity on Resource. A new entity-focused resource detection process can be created which allows add/override behavior at the entity level, rather than individual attribute level.
+
+For example, the existing resource detector looks like this:
+
+```yaml
+processors:
+  resourcedetection/docker:
+    detectors: [env, docker]
+    timeout: 2s
+    override: false
+```
+
+The future entity-based detector would look exactly the same, but interact with the entity model of resource:
+
+```yaml
+processor:
+  entityresourcedetection:
+     # Order determines override behavior
+     detectors: [env, docker]
+     # False means only append if entity doesn't already exist.
+     override: false 
+```
+
+The list of detectors is given in priority order (first wins, in event of a tie, outside of override configuration). The processor may need to be updated to allow the override flag to apply to each individual detector.
+
+The rules for attributes would follow entity merging rules, as defined for the SDK resource proivder.
+
+Note: While this proposals shows a new processor replacing the `resourcedetection` processor, the details of whether to modify-in-place the existing `resourcedetection` processor or create a new one would be determined as a follow up to this design. Ideally, we don't want users to need new configuration for resource in the otel collector.
+
+## Datamodel Changes
+
+Given our desired design and algorithms for detecting, merging and manipulating Entities, we need the ability to denote how entity and resource relate. These changes must not break existing usage of Resource, therefore:
+
+- The Entity model must be *layered on top of* the Resource model.  A system does not need to interact with entities for correct behavior.
```

Comment:
Does the opt-in nature here conflict with the goals of unblocking the pieces you mention at the start?  I'm wondering specifically about some components in a pipeline using the entities while others are only using resource.  In such a case, I would expect that two entities could appear in the pipeline because now we have two areas of reference for the same concept.

If entiies becomes a new scope, should this eventually seek to replace resource?

Reply:
I think entities will eventually replace existing resource usage.   The reason we're layering is to not break the entire ecosystem and effectively create a V2 of the protocol.  Instead, we see this as an evolution.  If you want to engage with Entities, you would NEED to ensure all components are entity aware.  As soon as one component is not, you're back to today's behavior.  So users who want to use entities MUST ensure all components leverage entities.

However - there's a higher level question.  Do I need to understand entities to interact with Metrics/Logs/Spans/etc. from OpenTelemetry?  The answer to that is no.  You can blindly leverage `Resource.attributes` as important slices of information for the data, and you will always be able to continue doing that.  That's addressing consumers / storage solutions.